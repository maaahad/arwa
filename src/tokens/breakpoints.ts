export enum Breakpoints {
  xs = '@media screen and (width < 575px)',
  sm = '@media screen and (width >= 575px)',
  md = '@media screen and (width >=768px)',
  lg = '@media screen and (width >= 992px)',
  xl = '@media screen and (width >= 1200px)',
  smOnly = '@media screen (575px <= width < 768px)',
  mdOnly = '@media screen and (768px <= width < 992px)',
  lgOnly = '@media screen and (992px <= width <1200px)',
}
