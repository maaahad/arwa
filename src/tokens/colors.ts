export enum Light {
    piccolo = '#4E46B4',
    hit = '#40A69F',
    beerus = '#EBEBEB',
    gohan = '#FFFFFF',
    goten = '#FFFFFF',
    goku = '#F5F5F5',
    bulma = '#000000',
    trunks = '#999CA0',
    popo = '#000000',
    jiren = "#e9e8f6",
    heles = "#f5f5f5",
    zeno = "#707070",
}

export enum Dark {
    piccolo = '#4E46B4',
    hit = '#0B0B0B',
    beerus = '#1F1F1F',
    gohan = '#1F1F1F',
    goten = '#FFFFFF',
    goku = '#0B0B0B',
    bulma = '#FFFFFF',
    trunks = '#999CA0',
    popo = '#000000',
    jiren = "#e9e8f6",
    heles = "#f5f5f5",
    zeno = "#707070",
}

export enum Supportive {    
    // krillin: Warning color
    krillin_100 = '#ffb319',
    krillin_60 = '#ffd47e',
    krillin_10 = '#fff9ed',
    // chichi: Error Color
    chichi_100 = '#ff4e64',
    chichi_60 = '#ff9ba8',
    chichi_10 = '#fff1f3',
    // roshi: Success color
    roshi_100 = '#2e7d32',
    roshi_60 = '#89b68c',
    roshi_10 = '#eef5ef',
    dodoria_100 = '#d33030',
    dodoria_60 = '#e68b8b',
    dodoria_10 = '#fceeee',
    cell_100 = '#95f1d5',
    cell_60 = '#c3f7e7',
    cell_10 = '#f7fefc',
    raditz_100 = '#b3804a',
    raditz_60 = '#d4b899',
    raditz_10 = '#f9f5f1',
    whis_100 = '#3448f0',
    whis_60 = '#8d98f7',
    whis_10 = '#eff0fe',
    frieza_100 = '#5c33cf',
    frieza_60 = '#a38ce4',
    frieza_10 = '#f2effb',
    nappa_100 = '#725550',
    nappa_60 = '#b09f9d',
    nappa_10 = '#f4f1f1',
}