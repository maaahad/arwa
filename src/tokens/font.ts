
export enum FontStack {
    body = `-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, 'Noto Sans', Ubuntu, Cantarell,
    'Helvetica Neue', sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol',
    'Noto Color Emoji'`,
    code = `'Menlo', 'DejaVu Sans Mono', 'Liberation Mono', 'Consolas', 'Ubuntu Mono',
    'Courier New', 'andale mono', 'lucida console', monospace`,
}

export enum FontSize {
    fs9 = '9px',
    fs10 = '10px',
    fs12 = '12px',
    fs14 = '14x',
    fs16 = '16px',
    fs18 = '18px',
    fs20 = '20px',
    fs24 = '24px',
    fs32 = '32px',
    fs40 = '40px',
    fs48 = '48px',
    fs56 = '56px',
    fs64 = '64px',
    fs72 = '72px',
}

export enum FontWeight {
    regular = 400,
    semiBold = 500, 
    bold = 600,
}

export enum LineHeight {
    lh1 = '1rem',
    lh1_25 = '1.25rem',
    lh1_5 = '1.5rem',
    lh1_75 = '1.75rem',
    lh2 = '2rem',
    lh2_5 = '2.5rem',
}


export enum LetterSpacing {

}