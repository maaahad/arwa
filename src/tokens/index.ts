export * as ColorTokens from "./colors";
export * as ShadowsTokens from "./shadows";
export * as FontTokens from "./font";
export * as ShapeTokens from "./shapes";
export * as BreakpointTokens from "./breakpoints";
export * as TransitionTokens from "./transitions";
