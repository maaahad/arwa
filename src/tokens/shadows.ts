
export enum Light {
    xs = '0 4px 12px -6px rgb(0 0 0 / 0.06)',
    sm = '0 6px 6px -6px rgb(0 0 0 / 0.16), 0 0 1px rgb(0 0 0 / 0.4)',
    md = '0 12px 12px -6px rgb(0 0 0 / 0.16), 0 0 1px rgb(0 0 0 / 0.4)',
    lg = '0 8px 24px -6px rgb(0 0 0 / 0.16), 0 0 1px rgb(0 0 0 / 0.4)',
    xl = '0 32px 32px -8px rgb(0 0 0 / 0.08), 0 0 32px -8px rgb(0 0 0 / 0.12), 0 0 1px rgb(0 0 0 / 0.2)',
}

// FIXME: This might need to be changed to something different
export enum Dark {
    xs = '0 4px 12px -6px rgb(0 0 0 / 0.06)',
    sm = '0 6px 6px -6px rgb(0 0 0 / 0.16), 0 0 1px rgb(0 0 0 / 0.4)',
    md = '0 12px 12px -6px rgb(0 0 0 / 0.16), 0 0 1px rgb(0 0 0 / 0.4)',
    lg = '0 8px 24px -6px rgb(0 0 0 / 0.16), 0 0 1px rgb(0 0 0 / 0.4)',
    xl = '0 32px 32px -8px rgb(0 0 0 / 0.08), 0 0 32px -8px rgb(0 0 0 / 0.12), 0 0 1px rgb(0 0 0 / 0.2)',
}