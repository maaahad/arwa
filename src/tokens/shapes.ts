export enum BorderRadius {
  xs = '4px',
  sm = '8px',
  md = '12px',
  lg = '16px',
}
