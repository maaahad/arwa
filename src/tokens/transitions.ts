export enum Transitions {
  arrival = "0.3s ease-out",
  departure = "0.2s ease-in",
}
