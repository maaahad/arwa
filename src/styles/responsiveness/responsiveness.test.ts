import { applyResponsiveness, getResponsiveCSSPropertyValue } from ".";
import { Theme } from "../themes";
import { BreakpointTokens } from "../../tokens";

describe("responsiveness", () => {
  test("getResponsiveCSSPropertyValue", () => {
    expect(getResponsiveCSSPropertyValue(100)).toBe("100px");
    expect(getResponsiveCSSPropertyValue("100%")).toBe("100%");
  });

  test("should get correct responsive CSS decleration", () => {
    const got = applyResponsiveness({
      theme: Theme.Light,
      width: [10, 20, 30, 40, 50],
      height: ["100px", 500, "100%"],
    });

    expect(got).toMatchSnapshot();
  });
});
