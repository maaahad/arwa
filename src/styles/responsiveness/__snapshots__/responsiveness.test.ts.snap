// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`responsiveness should get correct responsive CSS decleration 1`] = `
{
  "@media screen and (width >= 1200px)": {
    "width": "50px",
  },
  "@media screen and (width >= 575px)": {
    "height": "500px",
    "width": "20px",
  },
  "@media screen and (width >= 992px)": {
    "width": "40px",
  },
  "@media screen and (width >=768px)": {
    "height": "100%",
    "width": "30px",
  },
  "height": "100px",
  "width": "10px",
}
`;

exports[`responsiveness should get correct responsive CSS decleration 2`] = `
{
  "@media screen and (width >= 575px)": {
    "height": "500px",
  },
  "@media screen and (width >=768px)": {
    "height": "100%",
  },
  "height": "100px",
}
`;
